app/javascript/src/Navigation/components/header/SessionMenu.jsx

```diff
@@ -1,8 +1,11 @@
-import React from 'react'
+// @flow
+
+import * as React from 'react'
 import { Avatar } from '@patternfly/react-core'
 import avatarImg from 'Navigation/images/img_avatar.svg'
+import { AvatarProps } from 'Navigation/components/header/types'
 
-const SessionMenu = ({avatarLinkClass, impersonated, accountName, displayName, logoutPath, username}) => (
+const SessionMenu = ({avatarLinkClass, impersonated, accountName, displayName, logoutPath, username}: AvatarProps) => (
```

Comment:
`SessionMenuProps` would be better IMO or simply `Props` if we declare them inside this file. If needed somewhere else like Header.jsx it can be imported with a different name. I guess we fall again into personal preferences but this is what I'd do:
```js
// SessionMenu.jsx
export type Props = {
  avatarLinkClass: ...,
  impersonated: ...,
  ...
}

// Header.jsx
import type { Props as SessionMenuProps } from 'components/SessionMenu'
```

Reply:
I'd follow as a rule just `Props` when resides on the same file, and the type definition is just above the fn definition.

Reply:
When it's imported, then same name of the component + "Props" as in `ComponentProps`

Reply:
When it's imported, then same name of the component + "Props" as in `ComponentProps`